### ISSUE TITLE
Adding secret key (TOTP) to bitwarden password manager breaks Aegis auth code

### ISSUE THREAD

**Issue by Techy444** [beemdevelopment/Aegis#1059]
### Version

v2.1

### Source

F-Droid

### Vault encryption

Yes (with biometric unlock)

### Device

Pixel 6

### Android version

13

### ROM

GrapheneOS

### Steps to reproduce

1) Select one entry in Aegis app
2) Click on edit
3) copy secret key
4) add it to bitwarden password manager in the Authenticator Key (TOTP) field on windows desktop
5) Login using authentication code from Bitwarden and it will work
6) Login using authentication code from Aegis app and it will fail login

This occurs on any login you try, facebook, google, etc.

### What do you expect to happen?

1) Be able to login to my account using the authentication code from any source/device that gives me the authentication code

### What happens instead?

1) Login works using bitwarden generated authentication code and login fails using Aegis generated authentication code

**Comment by alexbakker** [comment]
Aegis doesn't know that you copied this entry to Bitwarden, Bitwarden doesn't know that you also have this entry in Aegis and the service that you've set up 2FA for (Google, Facebook, etc) doesn't know anything about this either. So this has to be something else. Probably a time synchronization issue. Is the clock on the device you're using Aegis on in sync? (Try navigating to https://time.is on that device to check)

Also, can you reproduce these reproduction steps reliably? Or did you just encounter this once?

**Comment by Techy444** [comment]
Hi Alex,

That's what I thought too, how could they know? I tried two or three different services yesterday. I would add the secret to bitwarden and only bitwarden auth codes would work. Then I removed it, then Aegis auth codes would work again.

But now I tried again with same services and different ones, and now both bitwarden and aegis auth codes work... However, bitwarden and aegis codes are do not match at any given time, yet both auth codes from either Aegis or Bitwarden works when logging in. Is that normal for the auth codes to be different but still work for logging in regardless?

**Comment by alexbakker** [comment]
> Is that normal for the auth codes to be different but still work for logging in regardless?

Yes. Usually, websites accept the current code and a couple of previous and future ones in order to account for the time being slightly out of sync on client devices. It's possible that Bitwarden has a different mechanism for obtaining the current time than Aegis. Aegis asks the system for the current time. So if either of those time sources is out of sync, that could result in Aegis and Bitwarden generating different codes.

> So maybe sometimes it works and sometimes it doesnt in TOR browser based on which location it picks?

Unlikely. This sounds like a time synchronization issue.
